benchmark_runner/common/oc/oc.py

```diff
@@ -441,6 +440,33 @@ def get_worker_nodes(self):
         """
         return self.run(fr""" {self.__cli} get nodes -l node-role.kubernetes.io/worker= -o jsonpath="{{range .items[*]}}{{.metadata.name}}{{'\n'}}{{end}}" """)
 
+    @staticmethod
+    @typechecked
+    def check_node_status(nodes_list: list):
+        """
+        This method check node status
+        @param nodes_list:
+        @return:
+        """
+        # Check if any node is not in 'Ready' status
+        for node in nodes_list:
+            node_name, node_status = node.split()
+            if node_status != 'Ready':
+                raise NodeNotReady(node_name, node_status)
+
+        # If no nodes are found in a non-ready state
+        return "All nodes are ready."
```

Comment:
Do you need to do this, or can you just return True?

Also, rather than raising an exception from here, perhaps return False, or a list of nodes that are not ready so the caller could decide what to do?

Reply:
Fixed